api/server.go

```diff
@@ -26,10 +29,17 @@ func writeJSON(w http.ResponseWriter, v interface{}) {
 
 type Wallet interface {
 	Balance() types.Currency
+	NextAddress() types.Address
+	Addresses() []types.Address
+	Transactions() []wallet.Transaction
+	FundTransaction(txn *types.Transaction, amount types.Currency, pool []types.Transaction) ([]types.ElementID, func(), error)
+	SignTransaction(vc consensus.ValidationContext, txn *types.Transaction, toSign []types.ElementID) error
 }
 
 type Server struct {
-	w Wallet
+	w  Wallet
+	s  *p2p.Syncer
```

Comment:
as with `Wallet`, we'll want to define a `Syncer` interface that `*p2p.Syncer` satisfies